### PR TITLE
[Feature] Build view individual permit holder page

### PR DESCRIPTION
**Pull request by j-bunk**
## Notion ticket link

[View individual Permit Holder page](https://www.notion.so/uwblueprintexecs/View-individual-Permit-Holder-page-396f2b4450ee4b30bab184678d5768da)

## Implementation description
Demo:
pr_demo

## Notes
* Making guardian information editable has been deprioritized for the MVP and will be addressed post-mvp so the edit guardian  modal was not built (clicking the edit button on the guardian information card does nothing)
* Alerts that pop up at the bottom after editing should be closable, I couldn't figure out how to make it closable and just removed the close button and added a ticket to the task board.
* The way data is handled in the components will likely change to match API response data

## Checklist
- [X] My PR name is descriptive, is in imperative tense and starts with one of the following: `[Feature]`,`[Improvement]` or `[Fix]`,
- [X] I have run the appropriate linter(s)
- [X] I have requested a review from the RCD team on GitHub, or specific people who are associated with this ticket

### CODE REVIEW

File: pages/admin/permit-holder/[permitHolderId].tsx
Comment (j-bunk) [inline pages/admin/permit-holder/[permitHolderId].tsx:82]: Temporary fix so that the CI build can pass. Will be fixed during API hookup since the way data is handled will change based on what data is received from the APIs

File: components/permit-holders/GuardianInformationCard.tsx
Comment (angeladietz) [inline components/permit-holders/GuardianInformationCard.tsx:19]: I think we should just remove this button temporarily since it doesn't do anything

File: components/permit-holders/AppHistoryCard.tsx
Comment (angeladietz) [inline components/permit-holders/AppHistoryCard.tsx:107]: For this table as well as the ones in Attached Files and Medical History, there's no divider between the header and data. I personally kinda like the divider line and it would also be more simple to leave it since they use the reusable table component, but we should get a designer's input to see if this is okay

File: components/permit-holders/MedicalHistoryCard.tsx
Comment (angeladietz) [inline components/permit-holders/MedicalHistoryCard.tsx:28]: (more of a design critique but) I think we should change "Disabling Condition(s)" to "Disabling Condition". Unless multiple conditions will go in the same row, we don't need it, and even in that case it still seems unnecessary to have that "(s)" there.

File: components/permit-holders/MedicalHistoryCard.tsx
Comment (angeladietz) [inline components/permit-holders/MedicalHistoryCard.tsx:37]: If this is a TODO for future dev work then I think you should add more context. Otherwise if you could resolve it that would be great

File: components/permit-holders/AppHistoryCard.tsx
Comment (angeladietz) [inline components/permit-holders/AppHistoryCard.tsx:77]: There's now an `ApplicationStatus` enum which was added by Oustan recently so it would be great if we could use that here instead. I know it needs to still be fixed in other places too so if its not trivial to only use it here then maybe don't worry about it and it can be fixed separately.

File: components/permit-holders/AppHistoryCard.tsx
Comment (angeladietz) [inline components/permit-holders/AppHistoryCard.tsx:35]: I think the status column seems too narrow on my screen, and I'm using a MacBook Air which has a fairly standard screen size. Is there any way the column widths could be adjusted to give more space between the status column and expiry date?
Screen Shot 2021-08-07 at 11 12 40 PM

File: components/permit-holders/PermitHolderHeader.tsx
Comment (angeladietz) [inline components/permit-holders/PermitHolderHeader.tsx:9]: This enum exists now (although I don't think it has all of these values... )

File: components/internal/PermitHolderInfoCard.tsx
Comment (angeladietz) [inline components/internal/PermitHolderInfoCard.tsx:23]: I think we can simplify this to: ` alignItems={alignGridItems || 'flex-start'}`

**Review by angeladietz — CHANGES_REQUESTED** [review]
Looks awesome! Just a couple small suggestions.

File: components/permit-holders/MedicalHistoryCard.tsx
Comment (j-bunk) [reply components/permit-holders/MedicalHistoryCard.tsx]: Just a note to myself that I forgot to remove, nice catch!

File: components/permit-holders/AppHistoryCard.tsx
Comment (j-bunk) [reply components/permit-holders/AppHistoryCard.tsx]: Agreed looks a bit too narrow on laptop screens, added 10px of padding to the right to address this
image

File: components/permit-holders/AppHistoryCard.tsx
Comment (j-bunk) [reply components/permit-holders/AppHistoryCard.tsx:77]: The enum Oustan added has different values than the values here (enum values: PENDING, APPROVED, REJECTED, COMPLETED). I think APPROVED == INPROGRESS and the rest are calculated based on the expiry date. 🤔 I'm not entirely sure how to handle that atm, i think i might just address it separately later

Comment (j-bunk) [reply components/permit-holders/AppHistoryCard.tsx:77]: cc @OustanDing

File: components/permit-holders/PermitHolderHeader.tsx
Comment (j-bunk) [reply components/permit-holders/PermitHolderHeader.tsx:9]: Related comment: https://github.com/uwblueprint/richmond-centre-for-disability/pull/44#discussion_r684750450

File: components/internal/RequestStatusBadge.tsx
Comment (j-bunk) [inline components/internal/RequestStatusBadge.tsx:18]: The `RequestStatusBadge` seem to sometimes resize based on the container they're in so I refactored it and specified the height and width explicitly

File: components/permit-holders/AppHistoryCard.tsx
Comment (j-bunk) [reply components/permit-holders/AppHistoryCard.tsx:107]: Yep good idea! I'll message the designers

File: components/permit-holders/MedicalHistoryCard.tsx
Comment (j-bunk) [reply components/permit-holders/MedicalHistoryCard.tsx:28]: So medical history shows the permit holder's medical conditions at the time of that application. Meaning the first row is all their medical conditions at the time of the most recent application, and the second row is all their medical conditions at the time of the application prior. So multiple conditions can go into one row, but I do agree though that the "(s)" isn't the most aesthetically pleasing. I'll bring it up with the designers and get their input

**Comment by j-bunk** [comment]
@angeladietz Thanks for all the great feedback! Addressed most of it and left comments for the ones I haven't fully addressed. I'll also message the designers to get their inputs on the points you brought up

File: components/internal/RequestStatusBadge.tsx
Comment (angeladietz) [reply components/internal/RequestStatusBadge.tsx:18]: Good catch! I think a simpler fix for this might have been to leave this file unchanged, except change the last return statement to this:

```
  return (
    <Wrap>
      <Badge variant={variant}>
        <Box ml="-4px" mr="8px">
          {_renderBadgeContent(variant)}
        </Box>
      </Badge>
    </Wrap>
  );
```

The only difference between this and your original code is the `<Wrap></Wrap>`, and this seemed to fix the issue as well. Feel free to stick to either solution, although this one would eliminate some of those hardcoded widths.

File: components/permit-holders/AppHistoryCard.tsx
Comment (j-bunk) [reply components/permit-holders/AppHistoryCard.tsx:107]: Andy said it's we're all good to just keep the divider for now

File: components/permit-holders/MedicalHistoryCard.tsx
Comment (j-bunk) [reply components/permit-holders/MedicalHistoryCard.tsx]: Dropped the "(s)" after getting Andy's input + approval

File: components/internal/RequestStatusBadge.tsx
Comment (j-bunk) [reply components/internal/RequestStatusBadge.tsx]: That looks cleaner so I implemented your suggestion!

I noticed that some positioning/spacing was slightly off so I tweaked the padding a bit to match the designs more closely
Before:
image

After minor padding tweaks (also modified one line from `badge.ts`):
Expiring  30 days

**Review by angeladietz — APPROVED** [review]
LGTM 🚀

**Pull request merged**
